Bencodex/Decoder.cs

```diff
@@ -279,7 +279,46 @@ private void Back()
             _offset--;
         }
 
-        private byte[] ReadDigits(bool takeMinusSign, byte delimiter)
+        // Reads the length portion for byte strings and unicode strings.
+        private int ReadLength()
+        {
+            const byte colon = 0x3a;    // ':'
+            const int asciiZero = 0x30; // '0'
+            int length = 0;
+
+            var b = ReadByte();
+
+            if (b is null)
+            {
+                throw new DecodingException(
+                    $"Expected digits, but the byte stream terminates at {_offset}.");
+            }
+
+            byte lastByte = b.Value;
+            while (lastByte != colon)
+            {
+#pragma warning disable SA1131
+                if (!(0x30 <= lastByte && lastByte <= 0x39)) // not '0'-'9'
```

Comment:
I agree that the current wording of this line, which checks the status of `lastByte`, could sometimes be easier to read.
And for optimization, it seems that we need to compute a little lighter.
```cs
if (lastByte < 0x30 || 0x39 < lastByte)
```
If `lastByte` is less than `0x30` or greater than `0x39`, I think the calculation will be a little faster, what do you think?

Reply:
If we want to squeeze every last bit of performance I suppose. Most intuitive form would actually be

```csharp
!(0x30 <= lastByte <= 0x39)
```

which isn't syntactically valid. 🙃

I do also prefer the form

```csharp
if (lastByte < 0x30 || 0x39 < lastByte)
```

as there is no compound logic. If people are okay with it, we could simply go with it. 😗

Reply:
Yeah, let's move forward. 😗